core/src/test/java/org/pentaho/di/connections/ConnectionManagerTest.java

```diff
@@ -256,6 +258,12 @@ public void testDefaultPropertiesNotNull() {
     Assert.assertNotNull( connectionDetails.getProperties().get( "baRoles" ) );
   }
 
+  @Test( expected = IllegalArgumentException.class )
+  public void testConnectionWithEmptyConnectionDetails() throws KettleException {
+    addOne();
```

Comment:
You also need a unit test for
connectionManager.test( nonNullConnectionDetails, null)

1. basically that an exception isn't thrown
2.  assert that `connectionProvider#test(ConnectionDetails,  VFSConnectionTestOptions)` gets called with an non null value


working example that tests for using mock verify:

- https://github.com/pentaho/pentaho-kettle/blob/627891d4c834d595a1f12fdd80bf4cb81b8298c1/core/src/test/java/org/pentaho/di/core/extension/ExtensionPointHandlerTest.java#L61
- https://github.com/pentaho/pentaho-scheduler-plugin/blob/1a7b1d13b13ab05dba32e3b9e8ae07e854f20948/core/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java#L288-L301
- however you would want to use Mockitos #isNotNull - https://javadoc.io/static/org.mockito/mockito-core/1.10.19/org/mockito/Matchers.html#isNotNull(java.lang.Class)


you can also go argument captor route: https://www.baeldung.com/mockito-argumentcaptor
- pentaho example: https://github.com/pentaho/pentaho-kettle/blob/627891d4c834d595a1f12fdd80bf4cb81b8298c1/engine/src/test/java/org/pentaho/di/trans/steps/regexeval/RegexEvalMetaTest.java#L80-L95

Reply:
FYI , Added the changes. As suggested in comment - https://github.com/pentaho/pentaho-kettle/pull/9305#discussion_r1595496535, added @NonNull and @Nullable annotations. After adding the annotations, the editor was complaining to remove IllegalArgumentException as the connectionDetails is never null. Hence removed and added objects.requireNonNull( connectionDetails );